src/skmultiflow/trees/extremely_fast_decision_tree.py

```diff
@@ -580,131 +578,7 @@ def _kill_subtree(self, node: AnyTimeSplitNode):
-        actual_model_size = calculate_object_size(self)
-        estimated_model_size = (self._active_leaf_node_cnt * self._active_leaf_byte_size_estimate
-                                + self._inactive_leaf_node_cnt * self._inactive_leaf_byte_size_estimate)
-        self._byte_size_estimate_overhead_fraction = actual_model_size / estimated_model_size
-        if actual_model_size > self.max_byte_size:
-            self.enforce_tracker_limit()
-
-    #  Override deactivate_all_leaves
-    def deactivate_all_leaves(self):
-        """ Deactivate all leaves. """
-        learning_nodes = self._find_learning_nodes()
-        for i in range(len(learning_nodes)):
-            if isinstance(learning_nodes[i], AnyTimeActiveLearningNode):
-                self._deactivate_learning_node(learning_nodes[i].node,
-                                               learning_nodes[i].parent,
-                                               learning_nodes[i].parent_branch)
-
-    # Override _deactivate_learning_node
-    def _deactivate_learning_node(self, to_deactivate: AnyTimeActiveLearningNode, parent: AnyTimeSplitNode,
-                                  parent_branch: int):
-        """ Deactivate a learning node.
-
-        Parameters
-        ----------
-        to_deactivate: AnyTimeActiveLearningNode
-            The node to deactivate.
-        parent:  AnyTimeSplitNode
-            The node's parent.
-        parent_branch: int
-            Parent node's branch index.
-
-        """
-        new_leaf = AnyTimeInactiveLearningNode(to_deactivate.get_observed_class_distribution())
-        if parent is None:
-            self._tree_root = new_leaf
-        else:
-            parent.set_child(parent_branch, new_leaf)
-        self._active_leaf_node_cnt -= 1
-        self._inactive_leaf_node_cnt += 1
-
-    # Override _activate_learning_node
-    def _activate_learning_node(self, to_activate: AnyTimeInactiveLearningNode, parent: AnyTimeSplitNode,
-                                parent_branch: int):
-        """ Activate a learning node.
-
-        Parameters
-        ----------
-        to_activate: AnyTimeInactiveLearningNode
-            The node to activate.
-        parent: AnyTimeSplitNode
-            The node's parent.
-        parent_branch: int
-            Parent node's branch index.
-
-        """
-        new_leaf = self._new_learning_node(to_activate.get_observed_class_distribution())
-        if parent is None:
-            self._tree_root = new_leaf
-        else:
-            parent.set_child(parent_branch, new_leaf)
-        self._active_leaf_node_cnt += 1
-        self._inactive_leaf_node_cnt -= 1
-
-    # Override measure_byte_size
-    def measure_byte_size(self):
-        """ Calculate the size of the tree.
-
-        Returns
-        -------
-        int
-            Size of the tree in bytes.
-
-        """
-        return calculate_object_size(self)
-
-    # Override rest
+    # Override reset
```

Comment:
Could you elaborate which are the methods you are referring to? 

The new `_new_learning_node` implementation was intended to enable `ExtremelyFastDecisionTreeClassifier` (and other tree algorithms) to use the base `enforce_tracker_limit`, `estimate_model_byte_size`, `_deactivate_learning_node`, and `_activate_learning_node` methods defined in `HoeffdingTreeClassifier`.

I may have let something pass by without realizing.

Reply:
Sorry for the confusion. My comment is about actual differences with the overridden methods in `ExtremelyFastDecisionTreeClassifier`. If the existing logic is the same as the inherited one, then let's remove it since it is indeed duplicated code. However, we must make sure that the code is indeed the same.

I am pinging @garawalid who implemented this algorithm for clarification.

Reply:
I checked `enforce_tracker_limit`, `estimate_model_byte_size`, `_deactivate_learning_node` and `_activate_learning_node` are the same in `HoeffdingTreeClassifier` and `ExtremelyFastDecisionTreeClassifier`. Also, the difference between the two algorithms is the way we build the tree and updates nodes.  
Also nodes of `ExtremelyFastDecisionTreeClassifier` inheret from `HoeffdingTreeClassifier`'s nodes.

So, this change looks good to me!

Reply:
Thanks for looking into this @garawalid